crates/bevy_asset/src/io/source.rs

```diff
@@ -443,10 +444,13 @@ impl AssetSource {
     /// the asset root. This will return [`None`] if this platform does not support writing assets by default.
     pub fn get_default_writer(
         _path: String,
-    ) -> impl FnMut() -> Option<Box<dyn AssetWriter>> + Send + Sync {
-        move || {
+    ) -> impl FnMut(bool) -> Option<Box<dyn AssetWriter>> + Send + Sync {
+        move |_should_create: bool| {
             #[cfg(all(not(target_arch = "wasm32"), not(target_os = "android")))]
-            return Some(Box::new(super::file::FileAssetWriter::new(&_path)));
+            return Some(Box::new(super::file::FileAssetWriter::new(
+                &_path,
+                _should_create,
+            )));
             #[cfg(any(target_arch = "wasm32", target_os = "android"))]
             return None;
         }
```

Comment:
I'm not a big fan of the `_` prefixed variable names. I know that they are not used in the `any(target_arch = "wasm32", target_os = "android")` cfg, but in that I case I think a `#[cfg_attr(any(target_arch = "wasm32", target_os = "android"), allow(unused_variables))]` will work better.

Reply:
this is the what's done in this module, I prefer to use the same in this PR

Reply:
fair enough.